test/test_pghoard.py

```diff
@@ -794,6 +795,14 @@ def test_pause_on_disk_full(self, db, pghoard_separate_volume, caplog):
         os.makedirs(wal_directory, exist_ok=True)
 
         pghoard.receivexlog_listener(pghoard.test_site, db.user, wal_directory)
+
+        # wait for pg_receivewal to start log streaming
+        # Otherwise, we might run into a race condition, where pg_receivewal will
+        # start streaming from the latest checkpoint instead of the first checkpoint
+        # this test actually generated
+        if 'starting log streaming' not in caplog.text:
+            time.sleep(0.1)
```

Comment:
**comment** For now I think waiting 100 milliseconds is enough. Otherwise, I can also add retries, but can make the test just a bit slower

Reply:
* Using the global log in a test for assertions (that is, assertions that are not about "this thing gets actually logged") is not recommended, as the log could change for reasons that are totally unconnected to this specific test and behaviour. We'd need a way to "directly" read the receivewal text somehow, if it's the only way to go forward.

BUT, I see it's used in other parts of this specific test, so I won't ask you to do this change rhrn.

* The risk here is sleeping forever and making this test hang if something goes wrong. You should adopt a time-based or iteration-based bailout; if after a certain time or number of attempts we don't get what we want, raise a TimeoutError (or anything else) and explain what we would have needed and it's not happening.